### PR TITLE
Provide reason for Output::Timeout

### PR DESCRIPTION
**Pull request by algesten**
This is by suggestion from @xnorpx.

### CODE REVIEW

**Review by davibe — APPROVED** [review]

**Comment by algesten** [comment]
My one concern is that it's a breaking change for the API, and maybe details that a new user of str0m might not need.

I'm weighing this API vs adding a `Rtc::last_timeout_reason()`, leaving the API as is, and giving us an API _if we want_ to know the reason.

**Comment by davibe** [comment]
> Rtc::last_timeout_reason()

like

**Comment by algesten** [comment]
Alright. Pushed the alternative API. Also fixed some timeouts that should have returned None.

File: src/lib.rs [hunk 1]
```diff
@@ -1466,9 +1548,34 @@ impl Rtc {
...
     }
 
+    /// The reason for the last [`Output::Timeout`]
+    ///
+    /// This is updated when calling [`Rtc::poll_output()`] and the next output
+    /// is a timeout.
+    ///
+    /// ```
+    /// # use str0m::{Rtc, Input, Output, Reason};
+    /// // A vec holding the managed rtc instances. One instance per remote peer.
```
Comment (xnorpx) [inline src/lib.rs:1563]: A vec comment can probably be removed here

**Comment by xnorpx** [comment]
Tested it seems to be working:

2024-05-25T15:51:07.036296Z ERROR something: Ice, Instant {
    tv_sec: 12329,
    tv_nsec: 182128520,
}
2024-05-25T15:51:09.489807Z ERROR something: Stats, Instant {
    tv_sec: 12331,
    tv_nsec: 636220215,
}

I like this change, it's easy for us to pull it if we want the information but doesn't brake the current API.

LGTM!

File: src/lib.rs (hunk 1 above)
Comment (algesten) [reply src/lib.rs]: thanks!

**Review by xnorpx — APPROVED** [review]
LGTM

**Pull request merged**